src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -864,36 +891,39 @@ public void testBasicDataSourceSerialization() throws Exception {
     // set up DataSource object and ensure connection works
     Map<String, String> params = getConnectionParameters();
     SnowflakeBasicDataSource ds = new SnowflakeBasicDataSource();
+    SnowflakeBasicDataSource ds2 = null;
     ds.setServerName(params.get("host"));
     ds.setSsl("on".equals(params.get("ssl")));
     ds.setAccount(params.get("account"));
     ds.setPortNumber(Integer.parseInt(params.get("port")));
     ds.setUser(params.get("user"));
     ds.setPassword(params.get("password"));
-    Connection con = ds.getConnection();
-    ResultSet resultSet = con.createStatement().executeQuery("select 1");
-    resultSet.next();
-    assertThat("select 1", resultSet.getInt(1), equalTo(1));
-    con.close();
-    File serializedFile = tmpFolder.newFile("serializedStuff.ser");
-    // serialize datasource object into a file
-    FileOutputStream outputFile = new FileOutputStream(serializedFile);
-    ObjectOutputStream out = new ObjectOutputStream(outputFile);
-    out.writeObject(ds);
-    out.close();
-    outputFile.close();
-    // deserialize into datasource object again
-    FileInputStream inputFile = new FileInputStream(serializedFile);
-    ObjectInputStream in = new ObjectInputStream(inputFile);
-    SnowflakeBasicDataSource ds2 = (SnowflakeBasicDataSource) in.readObject();
-    in.close();
-    inputFile.close();
+    try (Connection con = ds.getConnection();
+        Statement statement = con.createStatement();
+        ResultSet resultSet = con.createStatement().executeQuery("select 1")) {
```

Comment:
use statement to create resultset or you can just skip the `Statement statement = con.createStatement();` if you don't need the statement variable elsewhere here

Reply:
Same as above.